src/main/kotlin/no/skatteetaten/aurora/boober/feature/AuroraAzureAppFeature.kt

```diff
@@ -0,0 +1,95 @@
+package no.skatteetaten.aurora.boober.feature
+
+import com.fkorotkov.kubernetes.newObjectMeta
+import no.skatteetaten.aurora.boober.model.AuroraConfigFieldHandler
+import no.skatteetaten.aurora.boober.model.AuroraContextCommand
+import no.skatteetaten.aurora.boober.model.AuroraDeploymentSpec
+import no.skatteetaten.aurora.boober.model.AuroraResource
+import no.skatteetaten.aurora.boober.model.openshift.AuroraAzureApp
+import no.skatteetaten.aurora.boober.model.openshift.AzureAppSpec
+import no.skatteetaten.aurora.boober.service.AuroraDeploymentSpecValidationException
+import no.skatteetaten.aurora.boober.utils.boolean
+import no.skatteetaten.aurora.boober.utils.isListOrEmpty
+import no.skatteetaten.aurora.boober.utils.isValidDns
+
+val AuroraDeploymentSpec.azureAppFqdn: String?
+    get() {
+        return this.getOrNull(AuroraAzureAppFeature.ConfigPath.azureAppFqdn)
+    }
+
+val AuroraDeploymentSpec.azureAppGroups: List<String>?
+    get() {
+        //  Somewhat strange construction in order to avoid error with null as typeCast
+        val rawGroup = this.getOrNull<Any>(AuroraAzureAppFeature.ConfigPath.groups) ?: return null
+        return rawGroup as List<String>
+    }
+
+@org.springframework.stereotype.Service
+class AuroraAzureAppFeature : Feature {
+    object ConfigPath {
+        private const val root = "azure"
+        const val azureAppFqdn = "$root/azureAppFqdn"
+        const val groups = "$root/groups"
+    }
+
+    override fun enable(header: AuroraDeploymentSpec): Boolean {
+        return !header.isJob
+    }
+
+    override fun handlers(header: AuroraDeploymentSpec, cmd: AuroraContextCommand): Set<AuroraConfigFieldHandler> {
+        return setOf(
+            AuroraConfigFieldHandler(
+                ConfigPath.azureAppFqdn,
+                validator = { it.isValidDns(required = false) }
+            ),
+            AuroraConfigFieldHandler(
+                ConfigPath.groups,
+                validator = { it.isListOrEmpty(required = false) }
+            ),
+            AuroraConfigFieldHandler(
+                JwtToStsConverterFeature.ConfigPath.enabled,
+                defaultValue = false,
+                validator = { it.boolean(required = false) }
+            ),
+            AuroraConfigFieldHandler(JwtToStsConverterFeature.ConfigPath.version),
+            AuroraConfigFieldHandler(JwtToStsConverterFeature.ConfigPath.discoveryUrl),
+            AuroraConfigFieldHandler(JwtToStsConverterFeature.ConfigPath.ivGroupsRequired)
```

Comment:
Må disse settes her i tillegg til JwtToStsConverterFeature?

Reply:
Dersom jeg ikke har den, får jeg feil om jeg definerer opp JwtToStsConverterFeature. Feilen er at jeg da angir config-verdier som ikke finnes (for AuroraAzureAppFeature). Derfor tilføyes feltene, men uten validering her - i tillegg til i JwtToStsConverterFeature (som har validering). Om jeg ikke har dem med, får jeg "... is not a valid config field pointer"

Årsaken ligger i disse 2 featurene er innenfor samme struktur, f.eks:
```
azure:
  azureAppFqdn: dummy.test.local
  groups: []
  jwtToStsConverter:
    discoveryUrl: http://something.something/discovery/keys
    enabled: true
    version: 0.3.1
config:
```
Alternativet hadde vært å samle azure-ting i én Feature, men jeg synes ikke det blir lettere å forstå. Eksempel på det mønsteret er å finne i RouteFeature

Reply:
Skjønner. Kan vi gjøre det slik som dette?:

1. Opprette en ny pakke under features med navnet azure
2. Opprette en ny feature i den pakken med navnet AzureFeature som extender Feature.
3. Legge inn eksisterende azure features i samme pakke.
4. Ta bort at klassene extender Feature og bruker @Service
5. La AzureFeature bruke begge featurene, eks:
```
// AzureFeature
override fun generate(.....) {
 jwtToStsConverterFeature.generate()
 auroraAzureAppFeature.generate()
}
```